tools/gulp/tasks/library.ts

```diff
@@ -92,3 +101,23 @@ task('library:assets', ['library:assets:scss', 'library:assets:html']);
 task('library:assets:scss', sassBuildTask(materialDir, libraryRoot, true));
 task('library:assets:html', copyTask(join(libraryRoot, '**/*.+(html|scss)'), materialDir));
 task('library:assets:inline', () => inlineResources(materialDir));
+
+/**
+ * Finds the original sourcemap of the file and maps it to the current file.
+ * This is useful when multiple transformation happen (e.g TSC -> Rollup -> Uglify)
+ **/
+async function remapSourcemap(sourceFile: string) {
+  (await sorcery.load(sourceFile)).write();
```

Comment:
Yeah it recursively goes through the parent's sourcemap file. As from the package, it follows the chain of source-files back to the first one. But not worth mentioning how the package works, I think the JSDoc explains what it does.